File: src/paths.ts
import { homedir } from "node:os";
import { join } from "node:path";

const APP = "hideout";

function home(): string {
  return homedir();
}

export function configDir(): string {
  const raw = (process.env.XDG_CONFIG_HOME ?? "").trim();
  const base = raw ? raw : join(home(), ".config");
  return join(base, APP);
}

export function dataDir(): string {
  const raw = (process.env.XDG_DATA_HOME ?? "").trim();
  const base = raw ? raw : join(home(), ".local", "share");
  return join(base, APP);
}

export function configPath(): string {
  return join(configDir(), "config.json");
}

export function indexDir(): string {
  return join(dataDir(), "index");
}

export function pgliteDir(): string {
  return join(dataDir(), "pglite");
}

export function historyPath(): string {
  return join(dataDir(), "history");
}

export function setsPath(): string {
  return join(dataDir(), "sets.json");
}

File: src/hideout/paths.py
from __future__ import annotations

import os
from pathlib import Path

APP = "hideout"


def _home() -> Path:
    return Path.home()


def config_dir() -> Path:
    raw = os.environ.get("XDG_CONFIG_HOME", "").strip()
    base = Path(raw) if raw else _home() / ".config"
    return base / APP


def data_dir() -> Path:
    raw = os.environ.get("XDG_DATA_HOME", "").strip()
    base = Path(raw) if raw else _home() / ".local" / "share"
    return base / APP


def config_path() -> Path:
    return config_dir() / "config.toml"


def index_dir() -> Path:
    return data_dir() / "index"


def pglite_dir() -> Path:
    return data_dir() / "pglite"


def history_path() -> Path:
    return data_dir() / "history"


def sets_path() -> Path:
    return data_dir() / "sets.json"
